app/com/linkedin/drelephant/tuning/TuningInput.java

```diff
@@ -321,6 +336,53 @@ public void setScheduler(String scheduler) {
     return paramValueMap;
   }
 
+  @SuppressWarnings("unchecked")
+  public Map<String, Double> getSparkDefaultParams() throws JsonParseException, JsonMappingException, IOException{
+
+    ObjectMapper mapper = new ObjectMapper();
+    Map<String, Double> paramValueMap = new HashMap<String, Double>();
+    if (version == 1) {
+      paramValueMap = (Map<String, Double>) mapper.readValue(this._defaultParams, Map.class);
+    } else {
+      Map<String, String> paramsStringMap = (Map<String, String>) mapper.readValue(this._defaultParams, Map.class);
+      for (Map.Entry<String, String> entry : paramsStringMap.entrySet()) {
+        String confKey = entry.getKey();
+        String confVal = entry.getValue();
+        Double confValDouble = null;
+        try {
+          if (confKey.equals(SparkConfigurationConstants.SPARK_EXECUTOR_CORES_KEY)) {
```

Comment:
With so many else-ifs', it would be preferable to have a switch case instead

Reply:
Its legacy code , i'll take care of it next PR